### ISSUE TITLE
Speech to text model

### ISSUE THREAD

**Issue by hritikksingh** [hritikksingh/Twitter-video-emotion-and-sentiment-analysis#2]
Currently, we are using google speech recognition model to convert the speech into text but instead, if you are good at NLP you can work on a better speech to text model for Indian languages.
Requires knowledge of
- Python
- NLP

**Comment by Yara484** [comment]
I am a GSSoC participant, is this issue assigned yet? I code in python, did some research on NLP, and worked on small projects before so I think I can give this a try. Is understanding Indian languages a requirement?

**Comment by hritikksingh** [comment]
@Yara484 the main aim of this issue is to not use the Google speech recognition and make something of our own to reduce the dependency.
Can you create a Speech to text model? If yes then I will assign it to you.

**Comment by aritra1804** [comment]
Hey hrithik,can we use pretrained models for this. I guess we will receive better output with some fine tuning

**Comment by hritikksingh** [comment]
> Hey hrithik,can we use pretrained models for this. I guess we will receive better output with some fine tuning

I think we can use pre trained models, but it will increase dependencies. 
Go ahead and try it out, let's ee if that works out.

**Comment by aritra1804** [comment]
Thanks for assigning. Also for the dependency issue have u used virtual env?
I have seen this project has a lot of dependencies,ig  we can handle that with virtual env

**Comment by hritikksingh** [comment]
I didn't get you.
It's a module right so eitherr you are in venv or in main env, you have to install the modules right?

**Comment by aritra1804** [comment]
Yep but if we create a shared venv,the dependencies get reduced a lot

**Comment by hritikksingh** [comment]
> Yep but if we create a shared venv,the dependencies get reduced a lot

Okay we will try that.
On you after this issue.

**Comment by aritra1804** [comment]
great
also what is the deadline for this issue

**Comment by hritikksingh** [comment]
10 days should be enough time.

**Comment by aritra1804** [comment]
Great so 10th is my deadline

**Comment by anandxkumar** [comment]
Any update @aritra1804  ?

**Comment by aritra1804** [comment]
Hey, I am running a bit late since I had my endsems,could you give my 5 more days

**Comment by hritikksingh** [comment]
> Hey, I am running a bit late since I had my endsems,could you give my 5 more days

Sure

**Comment by hritikksingh** [comment]
@aritra1804 any update?

**Comment by aritra1804** [comment]
working on it, some issues with the training

**Comment by hritikksingh** [comment]
Okay, carry on buddy.
This issue takes time and effort.
All the best.

**Comment by hritikksingh** [comment]
@aritra1804 It's been so long, can you please update with your status.
And if you are no longer working on this then please let me know so that I can reassign this to someone else.

**Comment by aritra1804** [comment]
Hey @hritikksingh I m working on this, the approach I was trying initially had some problems so now i am trying another way. Almost done with it

**Comment by hritikksingh** [comment]
@aritra1804 Please make a PR as soon as you are done with it :)

**Comment by deepan773399** [comment]
please assign it to me .

**Comment by hritikksingh** [comment]
@aritra1804 please update your status on the issue or else I need to reassign.

**Comment by hritikksingh** [comment]
@aritra1804 please update by eod, else I will reassign.

**Comment by hritikksingh** [comment]
@deepan773399 I hope you have some plan to resolve this issue. I am assigning this issue to you.

**Comment by ghost** [comment]
Hi , iam with a good knowledge with python and Nlp I will use Google speech recognition,can you assign me this issue and thanks?

**Comment by hritikksingh** [comment]
Go ahead @tasneemahmed123
I hope you know what you have to do and there is no limitations, you can add anything that you find can add value.
Let is know if you have any query.

**Comment by navyasharma0203** [comment]
Can I work on this??

**Comment by hritikksingh** [comment]
@navyasharma0203 you can work on this but make sure you complete this before this month end.
And I hope you will be active on this :)

**Comment by hritikksingh** [comment]
@navyasharma0203 any update??

**Comment by navyasharma0203** [comment]
> @navyasharma0203 any update??

Yes...almost done...

**Comment by hritikksingh** [comment]
> > @navyasharma0203 any update??
> 
> Yes...almost done...

Great, try to raise a PR quick because we need to merge it before the program ends.
